backend/src/routes/TripRouter.ts

```diff
@@ -1,16 +1,86 @@
-import express from 'express';
-import { PrismaClient } from '@prisma/client';
+import express from "express";
+import { PrismaClient } from "@prisma/client";
 
 const router = express.Router();
 const prisma = new PrismaClient();
 
 // Get all trips
-router.get('/', async (req, res) => {
+router.get("/", async (req, res) => {
   try {
     const trips = await prisma.trip.findMany();
     res.json(trips);
   } catch (error) {
-    res.status(500).json({ error: 'An error occurred while fetching trips.' });
+    res.status(500).json({ error: "An error occurred while fetching trips." });
+  }
+});
+
+// Get an existing trip
+router.get("/:id", async (req, res) => {
+  const { id } = req.params;
+  try {
+    const trip = await prisma.trip.findUnique({
+      where: {
+        id,
+      },
+    });
+    res.status(200).json(trip);
+  } catch (error) {
+    res.status(500).json({ error: "An error occurred while fetching a trip." });
+  }
+});
+
+// Create a new trip
+router.post("/", async (req, res) => {
+  const { name, startDate, endDate, location } = req.body;
+  try {
+    const trip = await prisma.trip.create({
+      data: {
+        name,
+        startDate,
+        endDate,
+        location,
+      },
+    });
+    res.status(201).json(trip);
+  } catch (error) {
+    console.log(error);
+    res.status(500).json({ error: "An error occurred while creating the trip." });
+  }
+});
+
+// Update an existing trip
+router.put("/:id", async (req, res) => {
```

Comment:
Can `PUT `/trips/:id` add new participants or activities?

Reply:
PUT can add new participants and activities but we haven't specifed this yet for now